### PR TITLE
PronounDB: Add support for compact mode & clean up

### PR DESCRIPTION
**Pull request by TheKodeToad**
- More customisation
- Visible in compact mode!
- Less duplicate code.
- probably bad!
- made without coffee (i'm a tea person)!
- sorry romangraef if this conflicts :P.

### CODE REVIEW

File: src/plugins/pronoundb/components/PronounsChatComponent.tsx [hunk 1]
```diff
@@ -17,39 +17,63 @@
...
 
-import { fetchPronouns, formatPronouns } from "../pronoundbUtils";
-import { PronounMapping } from "../types";
+import { awaitAndFormatPronouns } from "../pronoundbUtils";
 
 const styles: Record<string, string> = findByPropsLazy("timestampInline");
 
-export default function PronounsChatComponentWrapper({ message }: { message: Message; }) {
+function shouldShow(message: Message): boolean {
+    // Respect showInProfile
```
Comment (Vendicated) [inline src/plugins/pronoundb/components/PronounsChatComponent.tsx:30]: outdated comment

File: src/plugins/pronoundb/components/PronounsProfileWrapper.tsx [hunk 2]
```diff
@@ -17,16 +17,19 @@
...
 import { UserStore } from "@webpack/common";
 
-import { fetchPronouns, formatPronouns } from "../pronoundbUtils";
-import { PronounMapping, UserProfilePronounsProps, UserProfileProps } from "../types";
+import { awaitAndFormatPronouns } from "../pronoundbUtils";
+import { UserProfilePronounsProps, UserProfileProps } from "../types";
 
 export default function PronounsProfileWrapper(PronounsComponent: React.ElementType<UserProfilePronounsProps>, props: UserProfilePronounsProps, profileProps: UserProfileProps) {
     const user = UserStore.getUser(profileProps.userId) ?? {};
+    // Respect showInMessages
```
Comment (Vendicated) [inline src/plugins/pronoundb/components/PronounsProfileWrapper.tsx:27]: same as above

File: src/plugins/pronoundb/index.ts [hunk 3]
```diff
@@ -30,15 +32,22 @@ export enum PronounsFormat {
...
 export default definePlugin({
     name: "PronounDB",
-    authors: [Devs.Tyman],
+    authors: [Devs.Tyman, Devs.TheKodeToad],
     description: "Adds pronouns to user messages using pronoundb",
     patches: [
+        {
+            find: "showCommunicationDisabledStyles",
+            replacement: {
+                match: /(\"span\",{id:.{1,2},className:.{1,2},children:.{1,2}}\))/,
```
Comment (Vendicated) [inline src/plugins/pronoundb/index.ts:41]:
```suggestion
                match: /(\"span\",{id:.\i,className:\i,children:\i}\))/,
```

File: src/plugins/pronoundb/styles.css [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.vc-pronoundb-compact {
+	display: none;
+}
+
+div[class*=compact] .vc-pronoundb-compact {
```
Comment (Vendicated) [inline src/plugins/pronoundb/styles.css:5]:
```suggestion
[class*=compact] .vc-pronoundb-compact {
```

**Review by Vendicated — CHANGES_REQUESTED** [review]

File: src/plugins/pronoundb/index.ts [hunk 5]
```diff
@@ -30,15 +32,23 @@ export enum PronounsFormat {
 
 export default definePlugin({
     name: "PronounDB",
-    authors: [Devs.Tyman],
+    authors: [Devs.Tyman, Devs.TheKodeToad],
     description: "Adds pronouns to user messages using pronoundb",
     patches: [
+        // Add next to username
```
Comment (Vendicated) [inline src/plugins/pronoundb/index.ts:38]: since it's now being added to both the username and timestamp, wouldn't that mean showing twice?

**Comment by Vendicated** [comment]
@TymanWasTaken Since it's your plugin, could you review please?

File: src/plugins/pronoundb/index.ts (hunk 5 above)
Comment (TheKodeToad) [reply src/plugins/pronoundb/index.ts:38]: The CSS fixes this. Well, the code still runs.

File: src/plugins/pronoundb/index.ts [hunk 6]
```diff
@@ -30,15 +32,23 @@ export enum PronounsFormat {
...
+        {
+            find: "showCommunicationDisabledStyles",
+            replacement: {
+                match: /(\"span\",{id:.\i,className:.\i,children:.\i}\))/,
```
Comment (Vendicated) [inline src/plugins/pronoundb/index.ts:42]: why are there 3 dots here? \i already covers that if you're just trying to match variable names

**Comment by Vendicated** [comment]
Doesn't actually seem to work?
image
image

**Comment by Vendicated** [comment]
Ah! It works after removing the periods mentioned in my recent feedback. Perhaps I gave you a wrong suggestion in code review, sorry if that was the case

**Review by Vendicated — APPROVED** [review]

**Pull request merged**